src/protocols/secure_channel/CASEServer.cpp

```diff
@@ -90,17 +87,32 @@ CHIP_ERROR CASEServer::OnMessageReceived(Messaging::ExchangeContext * ec, const
                                          const PayloadHeader & payloadHeader, System::PacketBufferHandle && payload)
 {
     ChipLogProgress(Inet, "CASE Server received SigmaR1 message. Starting handshake. EC %p", ec);
-    ReturnErrorOnFailure(InitCASEHandshake(ec));
-
-    mPairingSession.OnMessageReceived(ec, packetHeader, payloadHeader, std::move(payload));
+    CHIP_ERROR err = InitCASEHandshake(ec);
+    SuccessOrExit(err);
 
-    return CHIP_NO_ERROR;
     // TODO - Enable multiple concurrent CASE session establishment
+    // https://github.com/project-chip/connectedhomeip/issues/8342
+    ChipLogProgress(Inet, "CASE Server disabling CASE session setups");
+    mExchangeManager->UnregisterUnsolicitedMessageHandlerForType(Protocols::SecureChannel::MsgType::CASE_SigmaR1);
```

Comment:
So what happens here? That is, how is the next incoming message dispatched? What is seen by the initiator?

Reply:
The initiator will timeout, and needs to try again. This logic will be removed when #8342 is fixed. The current flow is causing racy behavior depending upon when the initiators are trying session setup.